src/Symfony/Component/Security/Core/User/CurrentUserProvider.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+namespace Symfony\Component\Security\Core\User;
+
+use Symfony\Component\Security\Core\Authentication\Token\Storage\TokenStorageInterface;
+
+/**
+ * Current User Provider.
+ *
+ * This provider gives you the current logged in user.
+ *
+ * @author Tobias Nyholm <tobias.nyholm@gmail.com>
+ */
+class CurrentUserProvider
+{
+    /**
+     * @var TokenStorageInterface tokenStorage
+     */
+    private $tokenStorage;
+
+    /**
+     * @param TokenStorageInterface $tokenStorage
+     */
+    public function __construct(TokenStorageInterface $tokenStorage)
+    {
+        $this->tokenStorage = $tokenStorage;
+    }
+
+    /**
+     * Get a user from the Security Token Storage.
+     *
+     * @return mixed
+     *
+     * @see TokenInterface::getUser()
+     */
+    public function getUser()
+    {
+        if (null === $token = $this->tokenStorage->getToken()) {
+            return;
+        }
+
+        if (!is_object($user = $token->getUser())) {
```

Comment:
I understand that some user representation might be a string. This PR is a refactor of the FrameworkBundle's `Controller::getUser`. I've just put it to a service. 
The `CurrentUserProvider::getUser` function do have the same logic as in the [FramworkBundle](https://github.com/symfony/symfony/blob/2.7/src/Symfony/Bundle/FrameworkBundle/Controller/Controller.php#L306)